pandas/sparse/array.py

```diff
@@ -322,31 +332,51 @@ def take(self, indices, axis=0):
             # return scalar
             return self[indices]
 
-        indices = np.atleast_1d(np.asarray(indices, dtype=int))
-
-        # allow -1 to indicate missing values
+        indices = com._ensure_platform_int(indices)
```

Comment:
sure! note I just merged the reorg of common.py (though it may not affect) as its just some import changes.